### PR TITLE
Using stash to backup Moodle data

### PR DESCRIPTION
**Pull request by danielpygo**
- stash, mysql, and stash operators installed
- steps.txt

### CODE REVIEW

File: examples/moodle-presslabs-stash/steps.txt [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    - Create a new course
+        - Upload an image
+
+2) See the composition of Moodle instance, checking for deployment name, should be moodle1
+    - kubectl get --raw "/apis/kubeplus.cloudark.io/v1/composition?kind=Moodle&instance=moodle1&namespace=namespace1" | python -mjson.tool
+
+3) Add a label to the Moodle deployment
+    - kubectl label -n namespace1 deployments moodle1 app=moodle-deploy
+
+4) Add base64 encoded access codes to s3-secret.yaml
```
Comment (devdattakulkarni) [inline examples/moodle-presslabs-stash/steps.txt]: Can you add exact command here.

File: examples/moodle-presslabs-stash/steps.txt [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+Recover a snapshot
+-----------
+
+1) Check snapshots, pick one
+    - kubectl get snapshots -n snapshot1 -l repository=deployment.moodle1
+
+2) Create PersistentVolumeClaim
+    - kubectl create -f pvc.yaml
+
+3) Change line 10 in recovery.yaml to the snapshot from step 9.
```
Comment (devdattakulkarni) [inline examples/moodle-presslabs-stash/steps.txt]: What is step 9 here?

File: examples/moodle-presslabs-stash/steps.txt (hunk 1 above)
Comment (danielpygo) [reply examples/moodle-presslabs-stash/steps.txt]: Sure!

File: examples/moodle-presslabs-stash/steps.txt (hunk 2 above)
Comment (danielpygo) [reply examples/moodle-presslabs-stash/steps.txt]: Meant to be step 1.

File: examples/moodle-presslabs-stash/steps.txt [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+3) Install presslabs-mysql-operator  and stash operator(note openapispec.json needs to be added to deploy with kubeplus)
+
+helm install https://github.com/cloud-ark/operatorcharts/blob/master/mysql-operator-0.2.5.tgz?raw=true
+helm install https://github.com/cloud-ark/operatorcharts/blob/master/stash-operator-chart-0.8.3.tgz?raw=true
+
+4) Change directory
+    - cd examples/moodle-presslabs-stash
+
+5) Deploy moodle operator
+    - kubectl apply -f moodle-operator.yaml
```
Comment (devdattakulkarni) [inline examples/moodle-presslabs-stash/steps.txt]: I see. You are deploying moodle-operator using KubePlus whereas other two Operators using Helm. Cool.

File: examples/moodle-presslabs-stash/steps.txt [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    - kubectl label -n namespace1 deployments moodle1 app=moodle-deploy
+
+4) Add base64 encoded access codes to s3-secret.yaml
+
+5) Apply secret
+    - kubectl apply -f s3-secret.yaml
+
+6) Start taking backups
+    - kubectl apply -f restic-moodle.yaml
+
```
Comment (devdattakulkarni) [inline examples/moodle-presslabs-stash/steps.txt]: Can you add steps (aws commands) to verify that the backup has been created.

Comment (danielpygo) [reply examples/moodle-presslabs-stash/steps.txt]: Sure

File: examples/moodle-presslabs-stash/moodle-backup/s3-secret.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: s3-secret
+  namespace: namespace1
+type: Opaque
+data:
+  AWS_ACCESS_KEY_ID: QUtJQVVMRktJTUNVRkhLS0VISDU=
```
Comment (devdattakulkarni) [inline examples/moodle-presslabs-stash/moodle-backup/s3-secret.yaml]: You can remove the values.

File: examples/moodle-presslabs-stash/steps.txt [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+     - kubectl create -f cluster1-secret.yaml
+     - kubectl create -f cluster1.yaml
+
+   - Wait till MysqlCluster pods are ready
+     - kubectl get pods -n namespace1
+
+   - Once MysqlCluster pods are ready, deploy Moodle
+     - kubectl create -f moodle1.yaml
+
+   - Wail till Moodle Pod is ready (It will take about 5/6 minutes for Moodle Pod to become ready)
```
Comment (devdattakulkarni) [inline examples/moodle-presslabs-stash/steps.txt]: Wait

File: examples/moodle-presslabs-stash/steps.txt [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+
+7. Validate that it is taking backups (may take one minute)
+    - aws s3 ls s3://stash-testing34/stash1/deployment/moodle1/
+
+
+Recover a snapshot
+-----------
+
+1) Check snapshots, pick one
+    - kubectl get snapshots -n snapshot1 -l repository=deployment.moodle1
```
Comment (devdattakulkarni) [inline examples/moodle-presslabs-stash/steps.txt]: The repository name comes from Recovery Custom Resource instance.

File: examples/moodle-presslabs-stash/steps.txt [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+6) Start taking backups
+    - kubectl apply -f restic-moodle.yaml
+
+7. Validate that it is taking backups (may take one minute)
+    - aws s3 ls s3://stash-testing34/stash1/deployment/moodle1/
+
+
+Recover a snapshot
+-----------
+
```
Comment (devdattakulkarni) [inline examples/moodle-presslabs-stash/steps.txt]: We will need to first cd into moodle-recovery folder, right?

cd moodle-recovery

Comment (danielpygo) [reply examples/moodle-presslabs-stash/steps.txt]: I am sending a new PR for this

Comment (devdattakulkarni) [reply examples/moodle-presslabs-stash/steps.txt]: Sounds good

**Pull request merged**